File: flut/.flutter/lib/flutter/rendering/object.dart
import 'package:flutter/rendering.dart';
import 'package:flut/flut/runtime.dart';
import 'package:flut/flut/object.dart';
import 'package:flut/flut/error.dart';

class FlutRenderObject with FlutRealtimeObject<RenderObject> {
  FlutRenderObject.createFromObject({
    required FlutRuntime runtime,
    required int oid,
    required RenderObject target,
  }) {
    initRealtimeFromObject(
      runtime: runtime,
      oid: oid,
      type: 'RenderObject',
      target: target,
    );
  }

  @override
  dynamic getRawProperty(String property) {
    switch (property) {
      case 'attached':
        return flutTarget.attached;
      case 'parent':
        return flutTarget.parent;
      case 'paintBounds':
        return flutTarget.paintBounds;
    }
    throw FlutUnknownPropertyException('RenderObject', property);
  }

  @override
  bool setProperty(String property, dynamic value) {
    throw FlutUnknownPropertyException('RenderObject', property);
  }

  @override
  dynamic callMethod(
    String method,
    List<dynamic> args,
    Map<String, dynamic> kwargs,
  ) {
    throw FlutUnknownMethodException(method);
  }
}

File: flut/flutter/rendering/object.py
from flut._flut.engine import FlutRealtimeObject


class RenderObject(FlutRealtimeObject):
    _flut_type = "RenderObject"

    @property
    def attached(self) -> bool:
        return bool(self._flut_get("attached"))

    @property
    def parent(self):
        return self._flut_get("parent")

    @property
    def paintBounds(self):
        return self._flut_get("paintBounds")
